### ISSUE TITLE
function_default_parameter_at_end should not trigger for `isolated` parameter.

### ISSUE THREAD

**Issue by Supereg** [realm/SwiftLint#5793]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Feature or Enhancement Proposal

Swift 6 introduced a new notation for inheriting the actor isolation from the call site (see [SE-0420](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0420-inheritance-of-actor-isolation.md)). The special parameter `isolated` is typically placed as the first argument of a method with the default value `#isolation`.

This violates the `function_default_parameter_at_end` rule and I think such cases should receive an exemption from the rule.
The [current implementation](https://github.com/realm/SwiftLint/blob/8241909addfd1a10cefde7d0619c4b9c16ba43b1/Source/SwiftLintBuiltInRules/Rules/Idiomatic/FunctionDefaultParameterAtEndRule.swift#L75-L97), only considers closures as special cases.

```swift
// warning: Function Default Parameter at End Violation: Prefer to locate parameters with defaults toward the end of the parameter list (function_default_parameter_at_end)
func foo(isolation: isolated (any Actor)? = #isolation, bar: String) {
    print("foo \(bar)")
}
```

Looking at the original proposal (see [SE-0313](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0313-actor-isolation-control.md#actor-isolated-parameters)), `isolated` parameter can occur at any position. Checking for the `isolated` attribute and a default value `#isolation` should be enough to detect isolation inheritance.

**Comment by SimplyDanny** [comment]
This should be hidden behind a new option which is off by default.

**Comment by Supereg** [comment]
> This should be hidden behind a new option which is off by default.

So the idea would be to add something like a `ignore_isolation_inheritance_parameter` configuration option for the `function_default_parameter_at_end` rule that is disabled by default?

**Comment by SimplyDanny** [comment]
> > This should be hidden behind a new option which is off by default.
> 
> So the idea would be to add something like a `ignore_isolation_inheritance_parameter` configuration option for the `function_default_parameter_at_end` rule that is disabled by default?

Yes, that'd be the idea. For the option name, we might prefer the plural `ignore_isolation_inheritance_parameters`. That's currently inconsistent throughout the configurations. Plural feels more natural though.
